shell/platform/darwin/ios/framework/Source/FlutterTextInputPlugin.mm

```diff
@@ -1852,6 +1852,30 @@ - (UITextRange*)characterRangeAtPoint:(CGPoint)point {
   return [FlutterTextRange rangeWithNSRange:fml::RangeForCharacterAtIndex(self.text, currentIndex)];
 }
 
+// Overall logic for floating cursor's "move" gesture and "selection" gesture:
+//
+// Floating cursor's "move" gesture takes 1 finger to force press the space bar, and then move the
+// cursor. The process starts with `beginFloatingCursorAtPoint`. When the finger is moved,
+// `updateFloatingCursorAtPoint` will be called. When the finger is released, `endFloatingCursor`
+// will be called. In all cases, we send the relative point to the framework, so that framework
+// can animate the floating cursor.
+//
+// Notice that during the move gesture, the framework only animate the cursor visually. It's only
+// after the gesture is complete, will the framework update the selection to the cursor's
+// new position (with zero selection length). This means during the animation, the visual effect
+// of the cursor is temporarily out of sync with the selection state in both framework and engine.
+// But it will be in sync again after the animation is complete.
+//
+// Floating cursor's "selection" gesture also starts with 1 finger to force press the space bar,
```

Comment:
I vaguely remember the "selection" gesture doesn't call `{start, update, end}FloatingCursor` for the 2nd finger and that was the reason the whole selection thing had to be implemented in the iOS embedder (I think we usually prefer implementing selection handling logic in the framework to avoid synchronization problems), maybe mention that if it isn't documented already?

Reply:
There is only one start,update,...,end for the whole sequence whether it's one or two fingers. We do continue to get update calls during the second finger movement, but no specific event when the second finger is pressed.

Reply:
yeah this is something i should include. good that you remember it.